src/lib.rs

```diff
@@ -191,7 +190,8 @@ struct WalkDirOptions {
     max_open: usize,
     min_depth: usize,
     max_depth: usize,
-    sorter: Option<Box<FnMut(&OsString,&OsString) -> Ordering + 'static>>,
+    sorter: Option<Box<FnMut((&OsStr, Option<Metadata>),
```

Comment:
Since we haven't got the `Metadata` on hand already I think we'll need to consider the implications of fetching it multiple times. It might be a non-issue, but it might also be expensive.

At the very least, I think it's worth pulling these arguments into a `struct`, something like:

```rust
pub struct DirEntrySort {
    file_name: OsString,
    metadata: Option<fs::Metadata>,
}

impl DirEntrySort {
    pub fn file_name(&self) -> &OsStr {
        &self.file_name
    }

    pub fn metadata(&self) -> Option<&fs::Metadata> {
        self.metadata.as_ref()
    }
}
```

cc: @burntsushi